crates/milli/src/search/new/geo_sort.rs

```diff
@@ -84,24 +98,32 @@ pub struct GeoSort<Q: RankingRuleQueryTrait> {
 
     cached_sorted_docids: VecDeque<(u32, [f64; 2])>,
     geo_candidates: RoaringBitmap,
+
+    // Limit the number of docs in a single bucket to avoid unexpectedly large overhead
+    max_bucket_size: u64,
+    // Considering the errors of GPS and geographical calculations, distances less than distance_error_margin will be treated as equal
+    distance_error_margin: f64,
 }
 
 impl<Q: RankingRuleQueryTrait> GeoSort<Q> {
     pub fn new(
-        strategy: Strategy,
+        parameter: &Parameter,
```

Comment:
```suggestion
        parameter: Parameter,
```

When a structure derive `Copy` it’s generally best to take it by copy instead of reference